SkyBlock/SKYBLOCK.SK/Events.sk

```diff
@@ -79,8 +79,26 @@ on EntityDamageByEntityEvent:
 		set {_allowed} to checkislandaccess({_player},{_l},"EntityDamageByEntityEvent-%{_type}%")
 		if {_allowed} is false:
 			cancel event
+	#
+	# > If the damage has been done by an arrow or snowball.
 	if "%{_player}%" is "arrow" or "snowball":
-		cancel event
+		#
+		# > Check, if the shooter is a player, allow skeletons to throw arrows like they want.
+		if {_player}.getShooter() is a player:
+			#
+			# > Check, if the shooter is allowed to shoot and check the location of the entity.
+			set {_allowed} to checkislandaccess({_player}.getShooter(),location of event.getEntity())
```

Comment:
Here, the type of the victim could be sent over. This way, the flags can be checked trough.